src/storage-client/src/storage_collections.rs

```diff
@@ -1921,20 +1922,20 @@ where
             .await?;
 
         // Reorder in dependency order.
-        to_register.sort_by_key(|(id, ..)| *id);
-
-        // Register tables first, but register them in reverse order since earlier tables
-        // can depend on later tables.
-        //
-        // Note: We could do more complex sorting to avoid the allocations, but IMO it's
-        // easier to reason about it this way.
-        let (tables_to_register, collections_to_register): (Vec<_>, Vec<_>) = to_register
-            .into_iter()
-            .partition(|(_id, desc, ..)| matches!(desc.data_source, DataSource::Table { .. }));
-        let to_register = tables_to_register
-            .into_iter()
-            .rev()
-            .chain(collections_to_register.into_iter());
+        #[derive(Ord, PartialOrd, Eq, PartialEq)]
+        enum DependencyOrder {
+            /// Tables should always be registered first, and large ids before small ones.
```

Comment:
OOC, do you remember while we sort tables this way?

Reply:
`alter table` creates new versions of the table, and has the old versions depend on the new versions. So it's important to create new versions first, I guess! (But I don't recall having seen this before, so that's just guessing from context.)

Reply:
This is the reason, yes